testutils/sampling_manager.go

```diff
@@ -33,14 +33,15 @@ func newSamplingManager() *samplingManager {
 }
 
 type samplingManager struct {
+	sync.Mutex
```

Comment:
I've seen pushback on this pattern from some people in the open source, which is fairly reasonable - by embedding the struct we're adding Lock/Unlock public methods to our type, whether it needs to expose those or not. So perhaps this is not worth changing.